File: examples/hooks/src/test/java/io/fluentlenium/examples/hooks/ExampleHook.java
package io.fluentlenium.examples.hooks;

import io.fluentlenium.core.FluentControl;
import io.fluentlenium.core.components.ComponentInstantiator;
import io.fluentlenium.core.hook.BaseHook;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.pagefactory.ElementLocator;

import java.util.function.Supplier;

public class ExampleHook extends BaseHook<ExampleHookOptions> {
    public ExampleHook(FluentControl control, ComponentInstantiator instantiator, Supplier<WebElement> elementSupplier,
                       Supplier<ElementLocator> locatorSupplier, Supplier<String> toStringSupplier, ExampleHookOptions options) {
        super(control, instantiator, elementSupplier, locatorSupplier, toStringSupplier, options);
    }

    @Override
    public void submit() {
        System.out.println(getOptions().getMessage() + ": before click!");
        super.submit();
        System.out.println(getOptions().getMessage() + "ExampleHook: after click!");
    }
}

File: fluentlenium-kotest/src/test/kotlin/org/fluentlenium/adapter/kotest/hooks/ExampleHook.kt
package org.fluentlenium.adapter.kotest.hooks

import org.fluentlenium.core.FluentControl
import org.fluentlenium.core.components.ComponentInstantiator
import org.fluentlenium.core.hook.BaseHook
import org.openqa.selenium.WebElement
import org.openqa.selenium.support.pagefactory.ElementLocator
import java.util.function.Supplier

class ExampleHook(
    control: FluentControl,
    instantiator: ComponentInstantiator,
    elementSupplier: Supplier<WebElement>,
    locatorSupplier: Supplier<ElementLocator>,
    toStringSupplier: Supplier<String>,
    options: ExampleHookOptions
) :
    BaseHook<ExampleHookOptions>(control, instantiator, elementSupplier, locatorSupplier, toStringSupplier, options) {
    override fun click() {
        super.click()
        clickCount++
    }

    companion object {
        var clickCount = 0
    }
}
